lib/matplotlib/colors.py

```diff
@@ -232,9 +232,43 @@ def is_color_like(c):
 
 
 def _has_alpha_channel(c):
-    """Return whether *c* is a color with an alpha channel."""
-    # 4-element sequences are interpreted as r, g, b, a
-    return not isinstance(c, str) and len(c) == 4
+    """Return whether *c* is a color with an alpha channel.
+
+    If *c* is not a valid color specifier, then the result is
+    undefined but will return False.
```

Comment:
```suggestion
    If *c* is not a valid color specifier, then the result is undefined.
```
Current return value `False` is an implementation detail. This must not be of interest for users because we claim the result is undefined. With "undefined" we reseve the right to change behavior anytime and therefore users cannot rely on the return value. It's thus better not to state it at all.